src/debugSession/BrightScriptDebugSession.ts

```diff
@@ -415,13 +415,23 @@ export class BrightScriptDebugSession extends BaseDebugSession {
         let error: Error;
         this.logger.log('[launchRequest] Packaging and deploying to roku');
         try {
+            const start = Date.now();
             const packageEnd = this.logger.timeStart('log', 'Packaging');
-            //build the main project and all component libraries at the same time
-            await Promise.all([
-                this.prepareMainProject(),
-                this.prepareAndHostComponentLibraries(this.launchConfiguration.componentLibraries, this.launchConfiguration.componentLibrariesPort)
-            ]);
-            packageEnd();
+            
+            // synchronously prepare the projects based on configuration
+            if (this.launchConfiguration.prepareProjectFilesSynchronously) {
```

Comment:
I'm not entirely sure why we would need this functionality? What is the problem this is trying to solve? If you have 1 main project, and 3 component libraries, why would building the main project first, then all component libraries in parallel solve anything? Can you explain the use case?